File: backend/app/routes/listaclientes.py
from flask import Blueprint, request, jsonify
from app.models import Cliente
from db import db

listaclientes_bp = Blueprint('listaclientes_bp', __name__)

@listaclientes_bp.route('/api/listaclientes', methods=['GET'])
def obtener_clientes():
    clientes=db.session.query(Cliente).all()
    clientes_lista=[
        {"id_cliente":h.id_cliente,
        "DNI":h.DNI,
        "nombre":h.nombre,
        "direccion":h.direccion,
        "telefono":h.telefono,
        "email":h.email,
        }
        for h in clientes
    ]
    return jsonify({"clientes":clientes_lista})

File: my-app/src/components/ListaClientes.js
import React, { useEffect, useState } from 'react';

const ListaClientes = () => {
  const [clientes, setClientes] = useState([]);

  useEffect(() => {
    fetch('api/listaclientes')
        .then(response=>response.json())
        .then(data=>{setClientes(data.clientes);})
        .catch(error => console.error('Error fetching data:', error));
    },[]);

  return (
    <div>
      <h2>Lista de Clientes</h2>
      <table>
        <thead>
          <tr>
            <th>ID</th>
            <th>Nombre</th>
            <th>DNI</th>
            <th>Telefono</th>
            <th>Dirección</th>
            <th>Email</th>
            {/* Agrega otros encabezados según los campos */}
          </tr>
        </thead>
        <tbody>
          {clientes.map(cliente => (
            <tr key={cliente.id_cliente}>
              <td>{cliente.id_cliente}</td>
              <td>{cliente.nombre}</td>
              <td>{cliente.DNI}</td>
              <td>{cliente.telefono}</td>
              <td>{cliente.direccion}</td>
              <td>{cliente.email}</td>
              {/* Agrega otras columnas según los campos */}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default ListaClientes;
